File: src/contests/cg/puzzle/haskell/google-interview---the-two-egg-problem.hs
-- 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
-- https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

main :: IO ()
main = do
  n <- readLn :: IO Integer
  print (head [k | k <- [0 ..], k * (k + 1) `div` 2 >= n])

File: src/contests/cg/puzzle/pascal/google-interview---the-two-egg-problem.pas
// 🎮 CodinGame Puzzle - google-interview---the-two-egg-problem
// https://www.codingame.com/training/hard/google-interview---the-two-egg-problem

{$mode objfpc}{$H+}
var
  n, k: int64;
begin
  readln(n);
  k := 0;
  while k * (k + 1) div 2 < n do
    k := k + 1;
  writeln(k);
end.
